src/main/java/io/pravega/connectors/flink/FlinkPravegaTableFactoryBase.java

```diff
@@ -0,0 +1,509 @@
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_TYPE());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_FROM());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_CLASS());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_SERIALIZED());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_TYPE());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_CLASS());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_SERIALIZED());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_DELAY());
+
+        // format wildcard
+        properties.add(FORMAT() + ".*");
+
+        return properties;
+    }
+
+    protected abstract String getVersion();
+
+    protected DescriptorProperties getValidatedProperties(Map<String, String> properties) {
+        final DescriptorProperties descriptorProperties = new DescriptorProperties(true);
+        descriptorProperties.putProperties(properties);
+
+        new SchemaValidator(true, false, false).validate(descriptorProperties);
+        new PravegaValidator().validate(descriptorProperties);
+
+        return descriptorProperties;
+    }
+
+    protected SerializationSchema<Row> getSerializationSchema(Map<String, String> properties) {
+        @SuppressWarnings("unchecked")
+        final SerializationSchemaFactory<Row> formatFactory = TableFactoryService.find(
+                SerializationSchemaFactory.class,
+                properties,
+                this.getClass().getClassLoader());
+        return formatFactory.createSerializationSchema(properties);
+    }
+
+    protected DeserializationSchema<Row> getDeserializationSchema(Map<String, String> properties) {
+        @SuppressWarnings("unchecked")
+        final DeserializationSchemaFactory<Row> formatFactory = TableFactoryService.find(
+                DeserializationSchemaFactory.class,
+                properties,
+                this.getClass().getClassLoader());
+        return formatFactory.createDeserializationSchema(properties);
+    }
+
+    protected FlinkPravegaTableSource createFlinkPravegaTableSource(Map<String, String> properties) {
+        final DescriptorProperties descriptorProperties = getValidatedProperties(properties);
+        final TableSchema schema = descriptorProperties.getTableSchema(SCHEMA());
+        final DeserializationSchema<Row> deserializationSchema = getDeserializationSchema(properties);
+
+        ConnectorConfigurations connectorConfigurations = new ConnectorConfigurations();
+        connectorConfigurations.parseConfigurations(descriptorProperties, ConnectorConfigurations.ConfigurationType.READER);
+
+        // create source from the reader builder by using the supplied properties
+        TableSourceReaderBuilder tableSourceReaderBuilder = new Pravega().tableSourceReaderBuilder();
+        tableSourceReaderBuilder.withDeserializationSchema(deserializationSchema);
+
+        if (connectorConfigurations.getUid().isPresent()) {
```

Comment:
This multiple `if` blocks to initialize `connectorConfigurations` may be encapsulated into a private method to ease readability.